### ISSUE TITLE
Home page log out

### ISSUE THREAD

**Issue by davidnicholls125** [johnvanbreda/naturespot-d8#29]
This morning I was logged out of the site when I opened up the computer - the first time this has happened for a long time. Whilst I logged in easily enough every time I return to the home page I'm logged out again. It seems to be just the home page that triggers logging out as I can navigate around other pages without a problem. Something must have changed to suddenly cause this problem, is it litespeed?

**Comment by davidnicholls125** [comment]
It appears that you are not really logged out when returning the home page as you can go to any other page and it says you are logged in again. It sounds like the home page is therefore a cached view of a non-logged in page??

**Comment by davidnicholls125** [comment]
Also noticed that:
https://www.naturespot.org.uk/home   works
https://www.naturespot.org.uk             doesn't work (ie displays logged out page)

**Comment by davidnicholls125** [comment]
It seems to have corrected now (not sure if you did anything?). We all experienced the same issue so it wasn't just me! We realised that you weren't actually logged out as you could navigate to other pages and were logged in again. I imagine it was showing a cached page?

**Comment by johnvanbreda** [comment]
@davidnicholls125 I did have a look at this yesterday and had the same problem. It was indeed serving the cached version from Litespeed Cache - which is supposed to not happen if you are logged in. If it happens again it might be worth asking Clook for help.

**Comment by davidnicholls125** [comment]
FYI this is happening again. I've contacted Clook as suggested.

**Comment by davidnicholls125** [comment]
I sent Clook the observations that there is a difference in the URL behaviour:
https://www.naturespot.org.uk/home works
https://www.naturespot.org.uk doesn't work (ie displays logged out page)
This is the reply:
Logging in using the second URL, I get redirected to this URL:
https://www.naturespot.org.uk/%7Enature50/user/login
With: this message:
Redirects to external URLs are not allowed by default, use \Drupal\Core\Routing\TrustedRedirectResponse for it.

Looks like you are defining ~nature50 where it should not be, we strongly suggest not using ~ for anything in URLs.

**Comment by johnvanbreda** [comment]
We don't intentionally use ~nature50 in the links anywhere. But, when we moved to the new server a while ago we were getting strange links containing ~nature50 in the menu items which at the time I diagnosed was a result of the site thinking it was running in a sub-folder called ~nature50 (which will be the name of the home directory of your account on the shared server). It seemed at the time to be a server setup issue but I found a way of fixing it by adding a redirect to your settings file, which effectively repairs broken links containins ~nature50.

Another possible area of contention is that after you log-in, I added code to the submit button which redirects you to <front> (the Drupal home page). I wonder if this code is clashing with LiteSpeed Cache - so I've removed it for now, so our login and redirection code is completely standard. However I still get the problem that logging in from https://www.naturespot.org.uk just takes you to https://www.naturespot.org.uk/home and logging in a second time from that page works.

**Comment by davidnicholls125** [comment]
It's not great to have to log in twice.... What do you suggest next?
(nb I'm away Monday until Friday so may be out of touch until then).

**Comment by johnvanbreda** [comment]
I've just disabled LSCache on the site and the login starts working again. I then turned it on again and it worked for a while - but only until I'd visited https://www.naturespot.org.uk/home as a non-logged in user. At this point, LSCache would have put the /home URL's page content in the cache, so future visits to the page serve up the cached version even though I am logged in. I think this proves that the LSCache module is breaking the Drupal login process.

I think we should ask Clook to look at this again - LSCache should not be serving up pages after you log in. If this gets us nowhere, then we can look for support direct from LiteSpeed.
